### ISSUE TITLE
Compiling issues

### ISSUE THREAD

**Issue by DigitalGabriele** [ivipsourcecode/DS-SLAM#9]
Hello, 

I am struggling with compiling DS-SLAM. 
I have a folder _uwsim_ws/share/install_ where I have ORB-SLAM2 which works fine. 
I installed DS-SLAM in _uwsim_ws/share/install_, but when I ran ./DS_SLAM_BUILD.sh, I received errors: 

![Screenshot 2019-03-26 12 15 27](https://user-images.githubusercontent.com/5788396/54992916-df4a1480-4fc0-11e9-8c5f-e41c6a62beb2.png)

**Comment by zoeyuchao** [comment]
It seems your path is wrong and use "sudo" because the error said "no permission "

**Comment by zmll** [comment]
@zoeyuchao  when i compiled  the libsegmentation file ,I encountered an error :
/home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/libsegmentation.cpp:68:19: error: no matching function for call to ‘caffe::Net<float>::Forward()’
     net_->Forward();
                   ^
In file included from /home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/../caffe-segnet/include/caffe/caffe.hpp:12:0,
                 from /home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/libsegmentation.hpp:37,
                 from /home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/libsegmentation.cpp:34:
/home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/../caffe-segnet/include/caffe/net.hpp:52:31: note: candidate: const std::vector<caffe::Blob<Dtype>*>& caffe::Net<Dtype>::Forward(const std::vector<caffe::Blob<Dtype>*>&, Dtype*) [with Dtype = float]
   const vector<Blob<Dtype>*>& Forward(const vector<Blob<Dtype>* > & bottom,
                               ^
/home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/../caffe-segnet/include/caffe/net.hpp:52:31: note:   candidate expects 2 arguments, 0 provided
/home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/../caffe-segnet/include/caffe/net.hpp:58:10: note: candidate: std::__cxx11::string caffe::Net<Dtype>::Forward(const string&, Dtype*) [with Dtype = float; std::__cxx11::string = std::__cxx11::basic_string<char>]
   string Forward(const string& input_blob_protos, Dtype* loss = NULL);
          ^
/home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/../caffe-segnet/include/caffe/net.hpp:58:10: note:   candidate expects 2 arguments, 0 provided
In file included from /home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/../caffe-segnet/include/caffe/caffe.hpp:7:0,
                 from /home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/libsegmentation.hpp:37,
                 from /home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/libsegmentation.cpp:34:
/home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/../caffe-segnet/include/caffe/blob.hpp: In instantiation of ‘std::__cxx11::string caffe::Blob<Dtype>::shape_string() const [with Dtype = float; std::__cxx11::string = std::__cxx11::basic_string<char>]’:
/home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/../caffe-segnet/include/caffe/blob.hpp:122:49:   required from ‘int caffe::Blob<Dtype>::CanonicalAxisIndex(int) const [with Dtype = float]’
/home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/../caffe-segnet/include/caffe/blob.hpp:73:37:   required from ‘int caffe::Blob<Dtype>::shape(int) const [with Dtype = float]’
/home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/../caffe-segnet/include/caffe/blob.hpp:151:17:   required from ‘int caffe::Blob<Dtype>::LegacyShape(int) const [with Dtype = float]’
/home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/../caffe-segnet/include/caffe/blob.hpp:135:51:   required from ‘int caffe::Blob<Dtype>::channels() const [with Dtype = float]’
/home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/libsegmentation.cpp:48:43:   required from here
/home/xh/dev/catkin_ws/src/DS-SLAM/Examples/ROS/ORB_SLAM2_PointMap_SegNetM/libsegmentation/../caffe-segnet/include/caffe/blob.hpp:57:23: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (int i = 0; i < shape_.size(); ++i) {
                       ^
CMakeFiles/segmentation.dir/build.make:62: recipe for target 'CMakeFiles/segmentation.dir/libsegmentation.cpp.o' failed
make[2]: *** [CMakeFiles/segmentation.dir/libsegmentation.cpp.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/segmentation.dir/all' failed
make[1]: *** [CMakeFiles/segmentation.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
